e2e/security/security.go

```diff
@@ -0,0 +1,218 @@
+
+package security
+
+import (
+	"os"
+	"testing"
+
+	"github.com/sylabs/singularity/e2e/internal/e2e"
+	"github.com/sylabs/singularity/internal/pkg/buildcfg"
+)
+
+type ctx struct {
+	env e2e.TestEnv
+}
+
+// testSecurityUnpriv tests the security flag fuctionality for singularity exec without elevated privileges
+func (c *ctx) testSecurityUnpriv(t *testing.T) {
+	tests := []struct {
+		name       string
+		image      string
+		action     string
+		argv       []string
+		opts       []string
+		expectExit int
+	}{
+		// taget UID/GID
+		{
+			name:       "Set_uid",
+			image:      c.env.ImagePath,
+			action:     "exec",
+			argv:       []string{"id", "-u", "|", "grep", "99"},
+			opts:       []string{"--security", "uid:99"},
+			expectExit: 255,
+		},
+		{
+			name:       "Set_gid",
+			image:      c.env.ImagePath,
+			action:     "exec",
+			argv:       []string{"id", "-g", "|", "grep", "99"},
+			opts:       []string{"gid:99"},
+			expectExit: 255,
+		},
+		// seccomp from json file
+		{
+			name:       "SecComp_BlackList",
+			image:      c.env.ImagePath,
+			action:     "exec",
+			argv:       []string{"mkdir", "/tmp/foo"},
+			opts:       []string{"--security", "seccomp:./security/testdata/seccomp-profile.json"},
+			expectExit: 159,
+		},
+		{
+			name:       "SecComp_true",
+			image:      c.env.ImagePath,
+			action:     "exec",
+			argv:       []string{"true"},
+			opts:       []string{"--security", "seccomp:./security/testdata/seccomp-profile.json"},
+			expectExit: 0,
+		},
+		// capabilities
+		{
+			name:       "capabilities_keep_true",
+			image:      c.env.ImagePath,
+			action:     "exec",
+			argv:       []string{"ping", "-c", "1", "8.8.8.8"},
+			opts:       []string{"--keep-privs"},
+			expectExit: 255,
+		},
+		{
+			name:       "capabilities_keep-false",
+			image:      c.env.ImagePath,
+			action:     "exec",
+			argv:       []string{"ping", "-c", "1", "8.8.8.8"},
+			expectExit: 2,
+		},
+		{
+			name:       "capabilities_drop",
+			image:      c.env.ImagePath,
+			action:     "exec",
+			argv:       []string{"ping", "-c", "1", "8.8.8.8"},
+			opts:       []string{"--drop-caps", "CAP_NET_RAW"},
+			expectExit: 2,
+		},
+	}
+
+	for _, tt := range tests {
+		optArgs := []string{}
+		optArgs = append(optArgs, tt.opts...)
+		optArgs = append(optArgs, c.env.ImagePath)
+		optArgs = append(optArgs, tt.argv...)
+
+		c.env.RunSingularity(
+			t,
+			e2e.AsSubtest(tt.name),
+			e2e.WithCommand("exec"),
+			e2e.WithArgs(optArgs...),
+			e2e.ExpectExit(tt.expectExit),
+		)
+
+	}
+}
+
+// testSecurityPriv tests security flag fuctionality for singularity exec with elevated privileges
+func (c *ctx) testSecurityPriv(t *testing.T) {
+	tests := []struct {
+		name       string
+		argv       []string
+		opts       []string
+		expectExit int
+	}{
+		// taget UID/GID
+		{
+			name:       "Set_uid",
+			argv:       []string{"id", "-u", "|", "grep", "99"},
+			opts:       []string{"--security", "uid:99"},
+			expectExit: 1,
```

Comment:
If anyone knows why this is exiting 1, then please let me know. Otherwise this is ready for review.